geoh5py/shared/cut_by_extent.py

```diff
@@ -0,0 +1,518 @@
+# ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+#  Copyright (c) 2025 Mira Geoscience Ltd.                                     '
+#                                                                              '
+#  This file is part of geoh5py.                                               '
+#                                                                              '
+#  geoh5py is free software: you can redistribute it and/or modify             '
+#  it under the terms of the GNU Lesser General Public License as published by '
+#  the Free Software Foundation, either version 3 of the License, or           '
+#  (at your option) any later version.                                         '
+#                                                                              '
+#  geoh5py is distributed in the hope that it will be useful,                  '
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of              '
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the               '
+#  GNU Lesser General Public License for more details.                         '
+#                                                                              '
+#  You should have received a copy of the GNU Lesser General Public License    '
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.           '
+# ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+import numpy as np
+
+from geoh5py.shared.utils import clean_extent_for_intersection
+
+
+@dataclass
+class Plane:
+    """
+    Represents a 3D plane with origin and basis vectors.
+
+    :param origin: Origin point of the plane coordinate system.
+    :param u_vector: U-axis basis vector.
+    :param v_vector: V-axis basis vector.
+    :param normal: Normal vector of the plane.
+    """
+
+    origin: np.ndarray
+    u_vector: np.ndarray
+    v_vector: np.ndarray
+    normal: np.ndarray
+
+    @classmethod
+    def from_quad_origin_last(cls, vertices: np.ndarray) -> Plane:
```

Comment:
Seems more natural to pass origin and an array of co-planar vertices in and let the caller sort out that the last vertex is the origin.

Reply:
The problem is that the origin is ALWAYS the last for geoimage.

So you would like to pass the index of the origin as an attribute?

`def from_quad_origin_last(cls, vertices: np.ndarray, origin:in = 3) -> Plane:`

Reply:
oh, ok, pass origin, u and v points. Yes, could be!